### PR TITLE
Convert to .taskcluster.yml v1

### PR DESCRIPTION
**Pull request by reuben**

### CODE REVIEW

**Comment by reuben** [comment]
Green PR group!

**Comment by reuben** [comment]
This should be ready for review now. I tried to keep changes as contained to .taskcluster.yml as possible to avoid having to modify extensively every single task template, but we may want to clean things up going forward.

File: .taskcluster.yml [hunk 1]
```diff
@@ -1,65 +1,92 @@
...
-          path: "/tmp/artifacts/"
-          expires: "{{ '7 days' | $fromNow }}"
-    # Each task also requires explicit metadata
-    metadata:
-      name: "DeepSpeech Decision Task"
-      description: "DeepSpeech Decision Task: triggers everything."
-      owner: "{{ event.head.user.email }}"
-      source: "{{ event.head.repo.url }}"
+          env:
+            TC_DECISION_SHA: ef67832e6657f43e139a10f37eb326a7d9d96dad
```
Comment (lissyx) [inline .taskcluster.yml:52]: not sure we still need this

File: .taskcluster.yml [hunk 2]
```diff
@@ -1,65 +1,92 @@
...
+            - "--login"
+            - "-cxe"
+            - >
+              echo "deb http://archive.ubuntu.com/ubuntu/ xenial-updates main" > /etc/apt/sources.list.d/xenial-updates.list &&
+              apt-get -qq update && apt-get -qq -y install git python3-pip curl sudo &&
+              adduser --system --home /home/build-user build-user &&
+              cd /home/build-user/ &&
+              echo -e "#!/bin/bash\nset -xe\nenv && id && mkdir ~/DeepSpeech/ && git clone --quiet ${metadata.github.clone_url} ~/DeepSpeech/ds/ && cd ~/DeepSpeech/ds && git checkout --quiet ${metadata.github.ref}" > /tmp/clone.sh && chmod +x /tmp/clone.sh &&
+              sudo -H -u build-user /bin/bash /tmp/clone.sh &&
+              sudo -H -u build-user --preserve-env /bin/bash /home/build-user/DeepSpeech/ds/taskcluster/tc-schedule.sh \
```
Comment (lissyx) [inline .taskcluster.yml:65]: if we pass env vars manually, we might be able to drop `--preserve-env`

**Review by lissyx — APPROVED** [review]

**Comment by lissyx** [comment]
LGTM, just make sure it works well on tags as well, but reading the code it seems okay

File: .taskcluster.yml (hunk 2 above)
Comment (reuben) [reply .taskcluster.yml:65]: I think I'm gonna move the variables to `env:` and keep `--preserve-env` here. It keeps the benefit of this approach, which is explicitly listing all the envvars the script depends on, but avoids all the repetition.

**Pull request merged**